javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -104,9 +132,11 @@ private static int treeSize(int branchCount, int depthShift) {
         } else if (n >= length()) {
             return empty();
         } else {
-            final NodeModifier drop = Arrays::copyDrop;
-            final Object[] root = modifyLeaf(array, depthShift, offset + n, drop, drop);
-            return new BitMappedTrie<>(root, offset + n, length() - n, depthShift);
+            final int index = offset + n;
+            final Object[] root = arePointingToSameLeaf(0, n)
+                                  ? array
+                                  : modifyLeaf(array, depthShift, index, Arrays::copyDrop, Identity);
+            return new BitMappedTrie<T>(root, index, length() - n, depthShift).collapseHeight();
```

Comment:
Valid question (and awesome new `GitHub` changes).
`compactHeight` is needed in cases when e.g. you have a tree of height `5` and you `drop` most of it, that could fit on a tree of e.g. height `2`. In that case, the root will have a single child, and its child will also have a single child. Maybe even the next one, but that cannot be guaranteed (imagine a `1`, prepended by `0` ... you will have two levels, though it could fit on a single one ... note also, that this doesn't escalate, the biggest difference because of these is one level difference).
In this case (i.e. after `drop` and `take`) there's just no point in keeping all those extra depth, hence the compactor :).